CMakeLists.txt

```diff
@@ -358,8 +358,11 @@ zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,
 include(cmake/extra_flags.cmake)
 
 zephyr_cc_option(-fno-asynchronous-unwind-tables)
-zephyr_cc_option(-fno-pie)
-zephyr_cc_option(-fno-pic)
+
+if(NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc-clang")
```

Comment:
I think there is a better way to do this using compile capabilities, so we do not have to do this ifdeffry here for each toolchain... , so this would be defined in cmake/compiler/...

Reply:
something along those lines....
```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef654819a2..7089eaab7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -358,8 +358,7 @@ zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,
 include(cmake/extra_flags.cmake)

 zephyr_cc_option(-fno-asynchronous-unwind-tables)
-zephyr_cc_option(-fno-pie)
-zephyr_cc_option(-fno-pic)
+zephyr_compile_options($<$<COMPILE_LANGUAGE:C>:$<TARGET_PROPERTY:compiler,no_pie_pic>>)

 if(CONFIG_THREAD_LOCAL_STORAGE)
 # Only support local exec TLS model at this point.
diff --git a/cmake/compiler/gcc/compiler_flags.cmake b/cmake/compiler/gcc/compiler_flags.cmake
index ac0724022b..52cae6e521 100644
--- a/cmake/compiler/gcc/compiler_flags.cmake
+++ b/cmake/compiler/gcc/compiler_flags.cmake
@@ -192,3 +192,7 @@ endif()

 # Compiler flag for disabling pointer arithmetic warnings
 set_compiler_property(PROPERTY warning_no_pointer_arithmetic "-Wno-pointer-arith")
+
+
+set_compiler_property(PROPERTY no_pie_pic -fno-pie)
+set_compiler_property(APPEND PROPERTY no_pie_pic -fno-pic)
diff --git a/cmake/compiler/xcc-clang/compiler_flags.cmake b/cmake/compiler/xcc-clang/compiler_flags.cmake
index 8b0a29a053..2f4c824e8f 100644
--- a/cmake/compiler/xcc-clang/compiler_flags.cmake
+++ b/cmake/compiler/xcc-clang/compiler_flags.cmake
@@ -12,3 +12,5 @@ if($ENV{XCC_NO_G_FLAG})
   # Clear the related flag(s) here so it won't cause issues.
   set_compiler_property(PROPERTY debug)
 endif()
+
+set_compiler_property(PROPERTY no_pie_pic "")
```

Reply:
@tejlmand  please confirm